src/Command/MakeMigrationCommand.php

```diff
@@ -0,0 +1,105 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Command;
+
+use Doctrine\Bundle\MigrationsBundle\Command\DoctrineCommand;
+use Symfony\Bundle\MakerBundle\ConsoleStyle;
+use Symfony\Bundle\MakerBundle\DependencyBuilder;
+use Symfony\Bundle\MakerBundle\Exception\RuntimeCommandException;
+use Symfony\Bundle\MakerBundle\Str;
+use Symfony\Bundle\MakerBundle\Validator;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+use Symfony\Component\Console\Input\ArgvInput;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Amrouche Hamza <hamza.simperfit@gmail.com>
+ */
+final class MakeMigrationCommand extends AbstractCommand
+{
+    protected static $defaultName = 'make:migrations';
+
+    protected function configure()
+    {
+        $this
+            ->setDescription('Creates a migration with the diff or an empty migration if there are no diffs')
+            ->addOption('empty', null, InputOption::VALUE_OPTIONAL, 'Generate an empty classes', 'false')
+            ->addOption('db', null, InputOption::VALUE_REQUIRED, 'The database connection to use for this command.')
+            ->addOption('em', null, InputOption::VALUE_OPTIONAL, 'The entity manager to use for this command.')
+            ->addOption('shard', null, InputOption::VALUE_REQUIRED, 'The shard connection to use for this command.')
+            ->setHelp(file_get_contents(__DIR__.'/../Resources/help/MakeMigrations.txt'))
+        ;
+    }
+
+    protected function writeNextStepsMessage(array $params, ConsoleStyle $io)
+    {
+        $io->text([
+            'Next: We just generated the migration',
+            'Find the documentation at <fg=yellow>http://symfony.com/doc/current/bundles/DoctrineMigrationsBundle/index.html</>'
+        ]);
+    }
+
+    protected function configureDependencies(DependencyBuilder $dependencies)
+    {
+        $dependencies->addClassDependency(
+            DoctrineCommand::class,
+            'migrations'
+        );
+    }
+
+    protected function getParameters(): array
+    {
+        return [];
+    }
+
+    protected function getFiles(array $params): array
+    {
+        return [];
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output)
```

Comment:
It's not really an intention for commands to override `execute()`. I totally see why you did it though - this generator is a little bit different :). What about this:

* Add a new `protected function generateCode()` in `AbstractCommand` and put the `$this->generator->generate()` code there by default.

* Override *that* method instead of `execute()` in this class.

Then we won't need to repeat things, like the success message at the bottom.